### PR TITLE
Add a more secure type validator to gpio module based on #911

### PR DESCRIPTION
**Pull request by DanielBallaSZTE**
The same conditions are present as in #980 .

IoT.js-DCO-1.0-Signed-off-by: Daniel Balla dballa@inf.u-szeged.hu

### CODE REVIEW

File: src/modules/iotjs_module_gpio.c [hunk 1]
```diff
@@ -23,13 +23,14 @@
 
 static void iotjs_gpio_destroy(iotjs_gpio_t* gpio);
 static iotjs_gpio_t* iotjs_gpio_instance_from_jval(const iotjs_jval_t* jgpio);
-IOTJS_DEFINE_NATIVE_HANDLE_INFO(gpio);
+IOTJS_DEFINE_NATIVE_HANDLE_INFO_THIS_MODULE(gpio);
```
Comment (LaszloLango) [inline src/modules/iotjs_module_gpio.c]: `IOTJS_DEFINE_NATIVE_HANDLE_INFO_THIS_MODULE(gpio_reqwrap);`

File: src/modules/iotjs_module_gpio.c [hunk 2]
```diff
@@ -269,13 +270,12 @@ JHANDLER_FUNCTION(GpioConstructor) {
 
 
 JHANDLER_FUNCTION(Write) {
-  DJHANDLER_CHECK_THIS(object);
+  JHANDLER_DECLARE_THIS_PTR(gpio, gpio);
```
Comment (LaszloLango) [inline src/modules/iotjs_module_gpio.c]: `JHANDLER_DECLARE_THIS_PTR(gpio_reqwrap, gpio);`

File: src/modules/iotjs_module_gpio.c [hunk 3]
```diff
@@ -292,13 +292,12 @@ JHANDLER_FUNCTION(Write) {
 
 
 JHANDLER_FUNCTION(Read) {
-  DJHANDLER_CHECK_THIS(object);
+  JHANDLER_DECLARE_THIS_PTR(gpio, gpio);
```
Comment (LaszloLango) [inline src/modules/iotjs_module_gpio.c]: ditto

File: src/modules/iotjs_module_gpio.c [hunk 4]
```diff
@@ -314,12 +313,11 @@ JHANDLER_FUNCTION(Read) {
 
 
 JHANDLER_FUNCTION(Close) {
-  DJHANDLER_CHECK_THIS(object);
+  JHANDLER_DECLARE_THIS_PTR(gpio, gpio)
```
Comment (LaszloLango) [inline src/modules/iotjs_module_gpio.c]: ditto

**Comment by yichoi** [comment]
LGTM

**Pull request merged**